airbyte-oauth/src/main/java/io/airbyte/oauth/flows/MicrosoftTeamsOAuthFlow.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.oauth.flows;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.oauth.BaseOAuth2Flow;
+import io.airbyte.protocol.models.OAuthConfigSpecification;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Supplier;
+import org.apache.http.client.utils.URIBuilder;
+
+public class MicrosoftTeamsOAuthFlow extends BaseOAuth2Flow {
+
+  private static final String fieldName = "tenant_id";
+  private static String tenantId;
```

Comment:
These should not be static variables in the class.

Please adapt the signature of the parent base methods accordingly to pass the input configuration from the connector around in the parent classes.

You might need to change the signature of `BaseOAuth2Flow.completeOAuthFlow` to also include the inputConfiguration.